src/app/api/course/route.ts

```diff
@@ -24,3 +28,25 @@ export async function POST(request: NextRequest) {
     return handleCommonErrors(error);
   }
 }
+
+export async function PUT(request: NextRequest) {
+  try {
+    const body = courseSchemaWithId.parse(await request.json());
+    const course = await prisma.course.findFirst({
+      where: { id: body.id },
+    });
+    if (!course) {
+      return errorResponse({
+        message: 'Course not found',
+        statusCode: StatusCodeType.NOT_FOUND,
+      });
+    }
+    await prisma.course.update({ where: { id: body.id }, data: body });
+    return successResponse({
+      message: 'Course succesfully updated!',
+      statusCode: StatusCodeType.OK,
```

Comment:
I'm not entirely sure whether this should be OK or CREATED in case of edit?

But we can device that its OK in our case, and just use it in PUT always

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201

Reply:
IMO the status code should be 200 or 204 because it just updates the existing recourse and does not create a new one

Reply:
Just passing by here. As far as I understand, it should be 204 as the resource already existed.

Reply:
IMO this depends if the response is going to return any content. 200 would be more appropriate if there is json returned and 204 when there if there is not.

Reply:
Yeah, 204 indicates there is no response. Should be more common with something like delete operation.